packages/sanity/src/core/form/inputs/PortableText/PortableTextInput.tsx

```diff
@@ -194,16 +203,28 @@ export function PortableTextInput(props: PortableTextInputProps) {
       const isObject = item.kind !== 'textBlock'
       let input: ReactNode
 
-      // Only render the input if the item is open or new
-      if (isObject && (item.member.open || !existingItem)) {
-        const inputProps = {...(props as FIXME), absolutePath: item.node.path}
-        input = <FormInput {...inputProps} />
+      if (isObject && (!existingItem || !isEqual(item.member, existingItem?.member))) {
```

Comment:
`isEqual(item.member, existingItem?.member)` will do a deep equality check - is that intentional? Given that these has (or at least _should_) already have gone through a reconciliation step that preserves object identity, a shallow equal should suffice, no?

Reply:
Doesn't seem like it. I don't get the preferred result with a shallow comparison of the members.
It works with shallow if I only compare the value. Probably validation, presence, or similar objects perhaps?

Reply:
Works great now, I can compare object equality!

Reply:
Feels great to get to the bottom of that one 😮‍💨